client/ayon_maya/api/lib.py

```diff
@@ -4544,3 +4557,85 @@ def get_scene_units_settings(project_settings=None)-> tuple[str, str]:
     linear_unit = scene_units.get("linear_units", "cm")
     angular_unit = scene_units.get("angular_units", "deg")
     return linear_unit, angular_unit
+
+
+def connect_texture_reference_objects(
+        texture_connections, nodes_by_id, tx_ref_nodes_by_id):
+    """Connect texture reference object nodes to the target object
+    nodes if there is one.
+
+    Args:
+        texture_connections (list): list of texture reference
+        objects connection data
+        nodes_by_id (dict): The dict with node ids.
+        tx_ref_nodes_by_id (dict): The dict with texture reference node ids.
+    """
+    # Compare loaded connections to scene.
+    for texture_connection in texture_connections:
+        source_node = next(
+            iter(tx_ref_nodes_by_id.get(
+            texture_connection["sourceID"], [])
+            ), None
+        )
+
+        target_node = next(
+            iter(nodes_by_id.get(texture_connection["destinationID"], [])
+            ), None
+        )
```

Comment:
This is wrong - I'd argue that if there are MULTIPLE targets - that each of them should be connected.